core/queryalgebra/evaluation/src/main/java/org/eclipse/rdf4j/query/algebra/evaluation/iterator/CrossProductIteration.java

```diff
@@ -67,4 +67,12 @@ protected BindingSet getNextElement()
 
 		return null;
 	}
+
+	@Override
+	protected void handleClose()
+		throws QueryEvaluationException
+	{
+		super.handleClose();
```

Comment:
@avataar thanks for applying the fix, and you're quite right that it should be done everywhere. @ansell  is in fact doing this as part of Pull Request #294 - which has turned into a general audit and standardization of the Iteration hierarchy.